the one that adds some more banners

sturobson opened the pull request:
We add 3 new 'alert banners' that can be used with a button to dismiss them.

sturobson committed:
adds alert banners

we add alert banners for success, warning, failure. Also creates an 'icon button' that has no general vf-button styles

sturobson committed:
updates README

updates to show that the new alert banners are available for use

sturobson commented:
@khawkins98 I don't know if you had time/want to add anything regarding JS here? I have left this as a 'BYOJS' to avoid any possible conflicts.

khawkins98 commented on components/vf-banner/vf-banner--states.scss:
```diff
@@ -0,0 +1,43 @@
+  color: set-ui-color(vf-ui-color--black);
+}
+
+.vf-banner--alert {
```
As .vf-banner--alert is the "general" class, it should come first in the Sass or else there are specificity issues:

![image](https://user-images.githubusercontent.com/928100/78368696-4cc15580-75c4-11ea-83bf-7d2d74687efb.png)

sturobson replied:
kinda related(?): I was wondering if we should phase out (don't laugh) `vf-banner--phase` and instead have it as `vf-banner--alert vf-banner--info` or something.

sturobson replied:
because we need a generic class for the padding (which `vf-banner--phase` has) but I think it's confusing to have `vf-banner--phase vf-banner--success`

sturobson replied:
I've included a `vf-banner--info` variant. Changed `vf-banner--alert vf-banner--alert` duplication so the red one is `vf-banner--danger`

khawkins98 replied:
That makes sense. 

On the `:hover` `:focus` I was thinking more accessibility if you miss the animation while tabbing or hovering, something like:

(tab first, then hover)

![Screen Shot 2020-04-03 at 21 48 48](https://user-images.githubusercontent.com/928100/78399568-68911f80-75f5-11ea-98fc-4135ba0a5608.gif)

```css
.vf-banner__button:focus {
    border: 1px solid white;
}

.vf-banner__button:hover {
    border-bottom: 1px solid white;
}
```

sturobson replied:
ok, so maybe for `:focus` but that border on `:hover` is a little EBI/VF1.x—y

Also - I’ll look at adding the text-crop to `vf-core` so we can make the box actually square for `:focus` because it looks a little ugly.

If we _really_ need something like a border bottom on `:hover` perhaps we need it to look more like a button than it does now (it has no styling from researching several design systms alerts styles).

I’ll try and do both - the text-crop so the square outline is actually _square_ and also an example of it being a button, perhaps something that’s inset - but that kinda goes against current aesthetics.

sturobson replied:
I mean, in saying some of that - the hover effect isn’t really part of the system either :sob: 

I’m sure there’s a good compromise out there :)

khawkins98 replied:
Thinking about this, why do we _have_ to use an icon? Just as well or better to use "Close" or "Hide". 

For example, opening www.gov.uk in an incognito. It's a longer notification, but (as usual) their use of text works better most other systems.

The text is only a few characters longer (plus more explicit) and will bypass issues discussed above -- it will also reinforce a principle of avoiding icons wherever possible.

sturobson replied:
I'm sure a banner will take be larger on a small viewport, and there would be less text for 'hide' rather than a `x` icon. 

re: hover/focus.

As it's a button there's the cursor change on hover (although touch devices will have to guess).

I've tidied up the `:focus` state so that it's a nice square box.

khawkins98 replied:
Probably more than fair for now. As we don't even have a real use case for dismissing notifications (beyond the we-probably-need-it from HCA), we're at risk of premature optimisation.

khawkins98 commented on components/vf-banner/vf-banner--states.scss:
```diff
@@ -0,0 +1,43 @@
+    transform-origin: center center;
+    transition: transform 250ms linear;
+  }
+  &:hover {
```
fun spin :)

khawkins98 replied:
Suggestion, we also have some stroke/shadow on hover to indicate activity. 

Also probably a `:focus` state.

Maybe the above is general enough that it should go on `. vf-button--icon`?

sturobson replied:
I've duplicated the hover spin for now.

sturobson replied:
Resolving this as we’ve moved it to another conversation in this PR

khawkins98 reviewed (commented):
Left a couple of comments. Also a thought, are these mixin classes, in that we should be able to add `vf-banner--alert` to  a `vf-banner--top`?

sturobson committed:
updates banners

changes alert Sass partial name, creates info variant that duplicates phase variant. Changes SVG title

sturobson committed:
updates banners

changes alert Sass partial name, creates info variant that duplicates phase variant. Changes SVG title

sturobson commented:
Perhaps we should 

- deprecate `vf-banner--phase` in place for `vf-banner--alert vf-banner--info`
- make use of the yaml to create the alert variants (colours and with/without buttons (I guess the option to not dismiss is only really option on the `--info` alert.

sturobson committed:
tiny little readme update

sturobson committed:
adds outline on focus, changes padding because of EM box

sturobson committed:
adds outline on focus, changes padding because of EM box

sturobson committed:
makes use of yml a little more

sturobson commented:
> Left a couple of comments. Also a thought, are these mixin classes, in that we should be able to add `vf-banner--alert` to a `vf-banner--top`?

We _should_ be able to. `vf-banner--alert` is only adding padding to `vf-banner__content` (which could be 'over engineering' (we could look at doing the padding as a default with custom properties)).

The colours from the alert type should work too. 

Things like this make me want to refactor everything to use CSS custom properties :sob:

sturobson commented:
I'll try and tidy this link `:visited` colouring too … but this is part of a bigger thing.

![Screenshot 2020-04-06 at 09 19 02](https://user-images.githubusercontent.com/925197/78537655-b7061000-77e7-11ea-9c55-ba805c4e8f56.png)

khawkins98 commented:
> I'll try and tidy this link :visited colouring too … but this is part of a bigger thing.

Part of me wonders if we should just go ahead and make the default for visited be a less obvious grey, adding purple visited links for `vf-content` 

That would probably require less overrides?

sturobson commented:
re: the `--phase` banner currently in place for www.embl.org - the HTML needs a tidy up really. 

```
<div class="vf-banner vf-banner--phase">
  <div class="vf-banner__content">
    <p class="vf-text-body--3"><a href="https://www.embl.org/news/lab-matters/corona-updates/">EMBL’s response to the coronavirus outbreak</a></p>
  </div>
</div>
```
should become –
```
<div class="vf-banner vf-banner--alert vf-banner--info">
  <div class="vf-banner__content">
    <p class="vf-banner__text">
      <a href="https://www.embl.org/news/lab-matters/corona-updates/">EMBL’s response to the coronavirus outbreak</a>
    </p>
  </div>
</div>
```

when this PR is merged and the code included in the www.embl.org repo.

sturobson commented:
> Part of me wonders if we should just go ahead and make the default for visited be a less obvious grey, adding purple visited links for vf-content

this is done in #835

sturobson commented:
I think we can merge this, `--phase` CSS is still in place so it won't break the existing uses.

khawkins98 commented:
> re: the --phase banner currently in place for www.embl.org - the HTML needs a tidy up really.

We'll get that for free as the notification banners will get updated as we move to the embl-notifcations.

sturobson commented:
> We'll get that for free as the notification banners will get updated as we move to the embl-notifcations.

Excellent.

![excellent](https://media.giphy.com/media/3ohA2Yh1puinrd2dzO/giphy.gif)
